controller/backup_controller.go

```diff
@@ -207,6 +207,19 @@ func getLoggerForBackup(logger logrus.FieldLogger, backup *longhorn.Backup) *log
 	)
 }
 
+func (bc *BackupController) isBackupBeingUsedForVolumeRestore(backupName, backupVolumeName string) error {
+	volumes, err := bc.ds.ListVolumesByBackupVolumeRO(backupVolumeName)
+	if err != nil {
+		return err
+	}
+	for _, v := range volumes {
+		if v.Status.RestoreRequired {
+			return fmt.Errorf("backup %v cannot be deleted due to the ongoing volume %v restoration", backupName, v.Name)
+		}
+	}
+	return nil
+}
```

Comment:
`backupName` is not used in this method, so this method is more about determining if the backup volume is being used to restore instead of the backup.

So basically this means any backups can't be deleted if there is a volume restoring from the corresponding backup volume. I thought the backup should be able to deleted individually, if it's not used. 

Derek mentioned the current file lock for backup volume on backup store is based on volume level. Therefore, this is actually the current design, but probably should consider if this is reasonable.

```go
func (lock *FileLock) canAcquire() bool {
	canAcquire := true
	locks := getLocksForVolume(lock.volume, lock.driver)
	file := getLockFilePath(lock.volume, lock.Name)
	log.WithField("lock", lock).Debugf("Trying to acquire lock %v", file)
	log.Debugf("backupstore volume %v contains locks %v", lock.volume, locks)

	for _, serverLock := range locks {
		serverLockHasDifferentType := serverLock.Type != lock.Type
		serverLockHasPriority := compareLocks(serverLock, lock) < 0
		if serverLockHasDifferentType && serverLockHasPriority && !serverLock.isExpired() {
			canAcquire = false
			break
		}
	}

	return canAcquire
}
```